cli/src/main.rs

```diff
@@ -1,19 +1,235 @@
-use std::{fs::File, io::Error};
+use clap::{Parser, Subcommand};
+use std::{
+    env,
+    path::{Path, PathBuf},
+    process::Command,
+};
 
-use clap::Parser;
+pub mod development {
+    tonic::include_proto!("development");
+}
+use development::development_client::DevelopmentClient;
+use development::{EchoReply, EchoRequest, Empty, Success};
+
+type DynError = Box<dyn std::error::Error>;
+
+#[derive(Debug, Subcommand)]
+enum Commands {
+    /// Build the entire Mycelia project
+    /// Shortcut for: `cargo build --workspace && cargo xtask build`
+    Build,
+    /// Start the Mycelia development server
+    Start {
+        /// The domain to listen on.
+        /// Default: localhost
+        /// TODO: add support to override (both here and in the development_server)
+        #[clap(short, long, default_value = "localhost")]
```

Comment:
Can we just code this to the IP it should be, to avoid doing a localhost lookup every time?